conans/__init__.py

```diff
@@ -13,7 +13,8 @@
 
 # complex_search: With ORs and not filtering by not restricted settings
 COMPLEX_SEARCH_CAPABILITY = "complex_search"
-SERVER_CAPABILITIES = [COMPLEX_SEARCH_CAPABILITY, ]
+REVISIONS_CAPABILITY = "revisions"
+SERVER_CAPABILITIES = [COMPLEX_SEARCH_CAPABILITY, REVISIONS_CAPABILITY]
```

Comment:
This new capability implies two things in the server:

a) Resolve a reference without revision ("#") to the latest revision found. 
b) When we ask for files of a reference, the server normally answer with a dict of `{"filename": "url to download"}`. But now, we need to know if a new reference (of a newer revision) has been resolved. So the output has to include the new reference: `{"reference": "lib/1.0@conan/stable#3", "files": {"filename": "url to download"}}`